Makefile

```diff
@@ -4,134 +4,115 @@ PYTHON := python3
 
 test-python: test-builtin test-suite
 
 coverage:
-	@echo "Generating coverage report..."
+	@echo "Generating coverage report for $(if $(device:-=),$(device),lightning.qubit) device:"
 	$(PYTHON) $(TESTRUNNER) $(COVERAGE)
-	pl-device-test --device lightning.qubit --skip-ops --shots=20000 $(COVERAGE) --cov-append
-	pl-device-test --device lightning.qubit --shots=None --skip-ops $(COVERAGE) --cov-append
+	pl-device-test --device $(if $(device:-=),$(device),lightning.qubit) --skip-ops --shots=20000 $(COVERAGE) --cov-append
+	pl-device-test --device $(if $(device:-=),$(device),lightning.qubit) --shots=None --skip-ops $(COVERAGE) --cov-append
 
 coverage-cpp:
-	@echo "Generating cpp coverage report in BuildCov/out .."
+	@echo "Generating cpp coverage report in BuildCov/out for $(if $(backend:-=),$(backend),lightning_qubit) backend"
 	rm -rf ./BuildCov
-	cmake pennylane_lightning/src -BBuildCov  -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=ON -DENABLE_COVERAGE=ON
+	cmake -BBuildCov  -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=ON -DENABLE_COVERAGE=ON -DPL_BACKEND=$(if $(backend:-=),$(backend),lightning_qubit)
 	cmake --build ./BuildCov
-	cd ./BuildCov; ./tests/pennylane_lightning_test_runner; \
-	lcov --directory . -b ../pennylane_lightning/src --capture --output-file coverage.info; \
+	cd ./BuildCov; for file in *runner ; do ./$file; done; \
+	lcov --directory . -b ../pennylane_lightning/core/src --capture --output-file coverage.info; \
 	genhtml coverage.info --output-directory out
 
+build:
+	rm -rf ./Build
+	cmake -BBuild -DENABLE_BLAS=ON -DENABLE_KOKKOS=ON -DENABLE_WARNINGS=ON -DPL_BACKEND=$(if $(backend:-=),$(backend),lightning_qubit)
+	cmake --build ./Build $(VERBOSE)
+
 test-cpp:
 	rm -rf ./BuildTests
-	cmake $(LIGHTNING_CPP_DIR) -BBuildTests -DBUILD_TESTS=ON
-	cmake --build ./BuildTests --target pennylane_lightning_test_runner
-	cmake --build ./BuildTests --target test
+	cmake -BBuildTests -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=ON -DENABLE_KOKKOS=ON -DENABLE_OPENMP=ON -DENABLE_WARNINGS=ON -DPL_BACKEND=$(if $(backend:-=),$(backend),lightning_qubit)
+ifdef target
+	cmake --build ./BuildTests $(VERBOSE) --target $(target)
+	OMP_PROC_BIND=false ./BuildTests/$(target)
+else
+	cmake --build ./BuildTests $(VERBOSE)
+	OMP_PROC_BIND=false cmake --build ./BuildTests $(VERBOSE) --target test
+endif
 
 test-cpp-blas:
 	rm -rf ./BuildTests
-	cmake $(LIGHTNING_CPP_DIR) -BBuildTests -DBUILD_TESTS=ON -DENABLE_BLAS=ON
-	cmake --build ./BuildTests --target pennylane_lightning_test_runner
-	cmake --build ./BuildTests --target test
-
-test-cpp-no-omp:
-	rm -rf ./BuildTests
-	cmake $(LIGHTNING_CPP_DIR) -BBuildTests -DBUILD_TESTS=ON -DENABLE_OPENMP=OFF
-	cmake --build ./BuildTests --target pennylane_lightning_test_runner
-	cmake --build ./BuildTests --target test
+	cmake -BBuildTests -DBUILD_TESTS=ON -DENABLE_BLAS=ON -DENABLE_WARNINGS=ON -DPL_BACKEND=$(if $(backend:-=),$(backend),lightning_qubit)
```

Comment:
Should we add --blas, --openmp options (maybe others) to control CMake variables (removing Makefile sections)?

Reply:
With a bit more flexibility, I think we could simplify the CI files, replacing many CMake commands with `make coverage --blas --openmp` for example, unless we really want to keep explicit.

Reply:
That sounds like a good idea, but I think we can come back to this later.